Database

jase-d opened the pull request:
Went with plan B and used a CSV file. The efficiency difference is pretty crazy, my original strategy couldn't get past 2M and was rough towards the end taking 15+ mins then crashing. Although a little tricky at first, highly recommend it if your computer doesn't have a high memory capacity like mine.

Currently only stress tested Postgres.

@hongmm @mbbph @ClaySamuelC

jase-d committed:
Starting script, need to adjust timing

jase-d committed:
Seed 25M into CSV file and easily transfer to Postgresql with COPY query
Total time under 10 mins until in db
Review count will be 50M split between two separate db instances

hongmm reviewed (approved):
@jase-d LGTM. Question, why do you need both psql1 and psql2 files? Also, it looks like you have run cassandra in your schema file for yous psql schema, does cassandra use the same sql language? Remember to add newlines at the bottom of each file too!

jase-d commented:
> @jase-d LGTM. Question, why do you need both psql1 and psql2 files? Also, it looks like you have run cassandra in your schema file for yous psql schema, does cassandra use the same sql language? Remember to add newlines at the bottom of each file too!

So in total I need 50M reviews. So my plan is to have two separate databases that will carry half the reviews in each. The second file will be connected to a separate ip/port once on an EC2 Micro Instance. Not sure if it's necessary, but I think it might help plus think it's just good practice for me to think of scaling horizontally now, even if it may not be the best solution (Will need to study the subject more).

Also, Cassandra uses Cassandra query language or CQL, and it's extremely similar to a SELECT query language or SQL with regards on how to insert. The only difference I've seen is that when retrieving you cannot use the keyword LIKE to compare, and possibly can't use join tables, which I am not planning on using.

Bottom spaces, will definitely do. Can you remind me of the importance of that? I think it's in the ABB style guide, and want to be consistent either way, but still curious if it has an advantage behind the scenes.

hongmm commented:
Your reasoning definitely makes sense. I was not aware that CQL was that similar to SQL,  I was not able to get enough research done yesterday on Cassandra so makes perfect sense to keep a similar schema. 

As far as I know, adding blank space is more to keep your code looking clean and a widespread standard across developers. It's similar to how in like Google Docs or Word docs there is some empty whitespace reserved at the top/bottom of the file, just makes it easier to read without feeling like everything is squished together. It is a common enough that github specifically notes this so it is probably a good idea to get in the habit of doing this (eslint catches this too but github specifically notes if you have a EOF line or not)

jase-d commented:
Thanks a lot for that explanation, I’ll have to make it a habit then!

On Fri, Oct 16, 2020 at 8:38 AM Matt Hong <notifications@github.com> wrote:

> Your reasoning definitely makes sense. I was not aware that CQL was that
> similar to SQL, I was not able to get enough research done yesterday on
> Cassandra so makes perfect sense to keep a similar schema.
>
> As far as I know, adding blank space is more to keep your code looking
> clean and a widespread standard across developers. It's similar to how in
> like Google Docs or Word docs there is some empty whitespace reserved at
> the top/bottom of the file, just makes it easier to read without feeling
> like everything is squished together. It is a common enough that github
> specifically notes this so it is probably a good idea to get in the habit
> of doing this (eslint catches this too but github specifically notes if you
> have a EOF line or not)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CAT-DAMON/reviews/pull/2#issuecomment-710121356>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQGUKKJNUFPO7UEVKYXKV2DSLBSIPANCNFSM4SS5MU5Q>
> .
>

jase-d committed:
Be able to connect and load 25M in each database;

Files are not the focus at this point, as of now copy and paste
the COPY query into the cassandra shell and run

jase-d reviewed (commented):
More work on the database branch going to look into a way to run the queries as a simple script from terminal for Cassandra instead of copying and pasting.

There may have been a loss of 5000 / 25M records with Cassandra's COPY method, I think it was the last 5000 going by the logs. I think there are ways to ensure no losses but I've heard that is a downfall of Cassandra.
@hongmm @mbbph @ClaySamuelC

hongmm reviewed (approved):
@jase-d I dont know much about the Cassandra stuff yet and how your setImmediate works but assuming that it behaves as you expect it to, I think the code LGTM! I can double check tomorrow or sometime later once I get into it but for now with the knowledge I have, code looks fine.

jase-d committed:
Pushing up

jase-d committed:
Quick Save

jase-d committed:
timer

jase-d committed:
Run Postgresql COPY query via script;

Also, added in Timer which is a constructor function with methods
to create a stopwatch easily for inline testing

jase-d committed:
Update the timer function for brevity

jase-d commented:
Hey all, with my last commit I added a timer function that I think could be cool to collaborate on with everyone. Right now it just has the method I mentioned earlier for 'seconds' in the meeting. But if it is helpful, I was think we could expand on it a bit and create a function that we could re-use in the long term by just cloning it into any project as a library. Feel free to check it out and use it, and collaborate!
@hongmm @mbbph @ClaySamuelC

mbbph commented on server/postgresql/psqlBulkLoad.js:
```diff
@@ -0,0 +1,27 @@
+  user: 'username',
+  host: 'localhost',
+  database: 'sdc',
+  password: 'password123',
```
ditto as above

mbbph commented on point_in_time_v1/timer.js:
```diff
@@ -0,0 +1,28 @@
+    return `FINISHED IN ${Math.floor((end - start) / 60)} MIN ${Math.floor((end  - start) % 60)} SECONDS`
+
+  };
+}
```
This is super neat! Packaging it up into a library like you said would be great practice

mbbph commented on server/postgresql/psql1.js:
```diff
@@ -0,0 +1,46 @@
+  user: 'username',
+  host: 'localhost',
+  database: 'sdc',
+  password: 'password123',
```
may be best practice to place pwd into gitignore

jase-d replied:
Thanks for catching that, I'll add that into my next commit.
